components/modal/header.js

```diff
@@ -0,0 +1,29 @@
+import IconButton from '../icon-button';
+import './style.scss';
+import { __ } from '@wordpress/i18n';
+
+const ModalHeader = ( { icon, title, onClose, closeLabel } ) => {
+	const label = closeLabel ? closeLabel : __( 'Close window' );
+
+	return (
+		<div
+			className={ 'components-modal__header' }
+		>
+			<div>
+				<span aria-hidden="true">
+					{ icon }
+				</span>
+				<h1 id="modalID" >
```

Comment:
Ideally, modals should have a visible title. If a `title` prop is passed, this should be the title referenced by aria-describedby`. In this case, developers should be allowed to set an ID that should be unique.
When a `title` prop is not passed, then the modal needs an `aria-label` that should be required and have a meaningful text.